src/linking/remove_internal_symbols.cpp

```diff
@@ -149,7 +150,7 @@ void remove_internal_symbols(
     else if(is_function)
     {
       // body? not local (i.e., "static")?
-      if(has_body && !is_file_local)
+      if(has_body || (!is_file_local && (config.main == symbol.name.c_str())))
```

Comment:
Yes, after discussing it with @thk123 it appears that I had understood the initial condition wrong, and thought that we should include a function "*if* it has a body or ...". Thomas let me know that this is wrong, and that if it has no body it's useless to us, so you're right. It will be updated accordingly.